### PR TITLE
Various build fixes

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
The wonderfully charismatic issue from @samueldr  https://github.com/NixOS/nixpkgs/issues/45960.
It was very compelling :smile: 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/networking/p2p/amule/default.nix [hunk 1]
```diff
@@ -22,6 +22,8 @@ mkDerivation rec {
     ++ lib.optional httpServer libpng
     ++ lib.optional client libX11;
 
+  patches = [ ./upnp-1.8.patch ];
```
[the remark this answers is not in the document]
Comment (worldofpeace) [reply pkgs/tools/networking/p2p/amule/default.nix]: Someone actually dropped it as an attachment in the upstream issue. 
That's not reliable place to fetch from obviously :stuck_out_tongue_winking_eye:

**Comment by worldofpeace** [comment]
@xeji 
Hmm @GrahamcOfBorg didn't do the thing.

**Comment by xeji** [comment]
Yes, Ofborg seems a little down today. Maybe you can ask around on IRC, I'm on the road right now.

**Comment by grahamc** [comment]
In the original comment there two spaces between the username and the buildcommand, which is apparently a bug in ofborg. However I deleted the extra space and it seems to be working:

https://logs.nix.ci/?attempt_id=3560f106-d074-4a29-85fa-8fcca91f6a4c&key=nixos%2Fnixpkgs.45967

**Comment by xeji** [comment]
Thanks! Let's try again, whit definitely only one space:

@GrahamcOfBorg build valadoc SkypeExport amule imgcat photoflow sayonara xfce4-13.xfce4-mixer

**Comment by worldofpeace** [comment]
I should probably restrict the platforms on the ones that failed on darwin right?

**Comment by xeji** [comment]
> I should probably restrict the platforms on the ones that failed on darwin right?
Well, that depends on whether they built before. 

- `imgcat` built on darwin before this PR, so you can't just disable it: https://hydra.nixos.org/jobset/nixpkgs/trunk/jobs-tab?filter=imgcat.
- `sayonara` never built on darwin in available hydra history: https://hydra.nixos.org/job/nixpkgs/trunk/sayonara.x86_64-darwin/all
- same for `photoflow`.

So: please enable imgcat on darwin again, and we'll look at the logs in more detail (I can't open the ofborg logs above).

**Comment by xeji** [comment]
@GrahamcOfBorg build imgcat photoflow sayonara

**Comment by xeji** [comment]
Don't be confused :smile:  We built multiple packages in one job, and ofborg reports "failure" if at least one fails. You have to look at the logs to know which. So imgcat wasn't the problem after all.

**Comment by xeji** [comment]
That's ok, we had a successful build before.

**Review by xeji — APPROVED** [review]
Thank you!

**Comment by xeji** [comment]
backported in f3702544270..00a43c05d9a

**Pull request merged**